File: CS/WinWebSolution.Module/BO.cs
using System;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    public interface ISupportDemoAction { }
    [DefaultClassOptions]
    [System.ComponentModel.DisplayName("I have the Demo Action")]
    public class DomainObject1 : BaseObject, ISupportDemoAction {
        public DomainObject1(Session session) : base(session) { }
    }
    [DefaultClassOptions]
    [System.ComponentModel.DisplayName("I have the Demo Action")]
    public class DomainObject2 : BaseObject, ISupportDemoAction {
        public DomainObject2(Session session) : base(session) { }
    }
    [DefaultClassOptions]
    [System.ComponentModel.DisplayName("I don't have the Demo Action")]
    public class DomainObject3 : BaseObject {
        public DomainObject3(Session session) : base(session) { }
    }
}

File: VB/WinWebSolution.Module/BO.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module

    Public Interface ISupportDemoAction
    End Interface

    <DefaultClassOptions>
    <System.ComponentModel.DisplayName("I have the Demo Action")>
    Public Class DomainObject1
        Inherits BaseObject
        Implements ISupportDemoAction

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
    End Class

    <DefaultClassOptions>
    <System.ComponentModel.DisplayName("I have the Demo Action")>
    Public Class DomainObject2
        Inherits BaseObject
        Implements ISupportDemoAction

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
    End Class

    <DefaultClassOptions>
    <System.ComponentModel.DisplayName("I don't have the Demo Action")>
    Public Class DomainObject3
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
    End Class
End Namespace
